cpp/include/cudf/io/detail/json.hpp

```diff
@@ -25,67 +25,26 @@
 
 #include <rmm/cuda_stream_view.hpp>
 
-// Forward declarations
-namespace arrow {
-namespace io {
-class RandomAccessFile;
-}
-}  // namespace arrow
-
 namespace cudf {
 namespace io {
 namespace detail {
 namespace json {
 
 /**
- * @brief Class to read JSON dataset data into columns.
+ * @brief Reads and returns the entire data set.
+ *
+ * @param[in] sources Input `datasource` objects to read the dataset from
```

Comment:
For my own edumacation, is the recommendation for `libcudf` to use `@param[in]` instead of `@param`? 
I rather like this convention to disambiguate input params from output, but I've seen reviewers recommend otherwise.

Reply:
I have no preference. I usually just follow the style of adjacent code. 🤷🏻‍♂️

Reply:
This came up some time ago, and IIRC the conclusion was to omit [in/out] when all parameters are inputs. I think the rest of libcudf does not add this tag, since output parameters are not commonly used.
IMO the point is: mark the output parameters as such, as it is unexpected for a parameter to be for output. Otherwise the tags are not that important.